### ISSUE TITLE
Add Bayesian interface to Turing.jl

### ISSUE THREAD

**Issue by RaphaelS1** [RaphaelS1/Survival.jl#20]

**Comment by sourish-cmi** [comment]
Do you want to add it to Turing? And then just call it in `SurvivalAnalysis.jl`? 

Or do you want to develop it as part of `SurvivalAnalysis.jl` with `Turing.jl`?

**Comment by RaphaelS1** [comment]
Hey so that's an open question into what the integration effectively looks like. My current thinking is maybe a bridge package like SurvivalBayesian.jl which depends on both SurvivalAnalysis.jl and Turing.jl and basically is just glue code to build models

**Comment by sourish-cmi** [comment]
My only concern is that if we build too many packages, each package's maintenance could be a headache. So `SurvivalBayesian.jl` could be part of your `SurvivalAnalysis.jl`. Of course, it will depend on both `SurvivalAnalysis.jl` and `Turing.jl`. We can simply use `include(SurvivalBayesian.jl)` in the main module of  `SurvivalAnalysis.jl`. Let me know your thought.

**Comment by RaphaelS1** [comment]
Agreed - I'll give some more background tomorrow and then summarise our chat here

**Comment by sourish-cmi** [comment]
Wonderful. Looking forward to our discussion tomorrow.

**Comment by RaphaelS1** [comment]
Thanks also adding [this](https://discourse.julialang.org/t/question-about-bayesian-survival-analysis-with-cox-proportional-hazards-regression-method-in-julia-with-turing/80672/3) discussion from discourse
